pkg/jwt/jwt.go

```diff
@@ -83,11 +90,25 @@ func (t *TokenService) DecodeToken(token string) (*TokenContext, error) {
 		UserID:         claims["UserID"].(string),
 		UserName:       claims["UserName"].(string),
 		UserEmail:      claims["UserEmail"].(string),
-	}, nil
+	}
+
+	if userGroups, ok := claims["UserGroups"]; ok && userGroups != nil {
+		if userGroupsSlice, ok := userGroups.([]any); ok {
+			groups := make([]string, 0, len(userGroupsSlice))
+			for _, group := range userGroupsSlice {
+				if groupStr, ok := group.(string); ok {
+					groups = append(groups, groupStr)
+				}
+			}
+			context.UserGroups = groups
```

Comment:
All of this makes me want to make the `UserGroups` claim a comma-delimited string to avoid all the casting. What are your thoughts?

Reply:
Had the same thought. I'm going to go ahead and do that instead.

Reply:
done